x/concentrated-liquidity/lp_test.go

```diff
@@ -2,66 +2,252 @@ package concentrated_liquidity_test
+			// for withdrawing a position.
+			sutConfigOverwrite: &lpTest{
+				lowerTick:     -1, // valid tick at which no position does not exist
+				expectedError: types.PositionNotFoundError{PoolId: 1, LowerTick: -1, UpperTick: 86129},
+			},
+		},
+		"error: pool id for pool that does not exist": {
+			// setup parameters for creation a pool and position.
+			setupConfig: baseCase,
+
+			// system under test parameters
+			// for withdrawing a position.
+			sutConfigOverwrite: &lpTest{
+				poolId:        2, // does not exist
+				expectedError: types.PoolNotFoundError{PoolId: 2},
+			},
+		},
+		"error: invalid tick given": {
+			// setup parameters for creation a pool and position.
+			setupConfig: baseCase,
+
+			// system under test parameters
+			// for withdrawing a position.
+			sutConfigOverwrite: &lpTest{
+				lowerTick:     types.MaxTick + 1, // invalid tick
+				expectedError: types.InvalidTickError{Tick: types.MaxTick + 1, IsLower: true},
+			},
+		},
+		"error: insufficient liqudity": {
+			// setup parameters for creation a pool and position.
+			setupConfig: baseCase,
+
+			// system under test parameters
+			// for withdrawing a position.
+			sutConfigOverwrite: &lpTest{
+				liquidityAmount: baseCase.liquidityAmount.Add(sdk.OneDec()), // 1 more than available
+				expectedError:   types.InsufficientLiquidityError{Actual: baseCase.liquidityAmount.Add(sdk.OneDec()), Available: baseCase.liquidityAmount},
+			},
+		},
+		// TODO: test with custom amounts that potentially lead to truncations.
+	}
+
+	for name, tc := range tests {
+		s.Run(name, func() {
+			var (
+				tc                 = tc
+				config             = *tc.setupConfig
+				sutConfigOverwrite = *tc.sutConfigOverwrite
+			)
+
+			s.SetupTest()
+
+			var (
+				ctx                         = s.Ctx
+				concentratedLiquidityKeeper = s.App.ConcentratedLiquidityKeeper
+				liquidityCreated            = sdk.ZeroDec()
+				owner                       = s.TestAccs[0]
+			)
+
+			// Setup.
+			if tc.setupConfig != nil {
+				_, err := concentratedLiquidityKeeper.CreateNewConcentratedLiquidityPool(ctx, config.poolId, denom0, denom1, config.currentSqrtP, config.currentTick)
+				s.Require().NoError(err)
+
+				_, _, liquidityCreated, err = concentratedLiquidityKeeper.CreatePosition(ctx, config.poolId, owner, config.amount0Desired, config.amount1Desired, sdk.ZeroInt(), sdk.ZeroInt(), config.lowerTick, config.upperTick)
+				s.Require().NoError(err)
+			}
+
+			mergeConfigs(&config, &sutConfigOverwrite)
+
+			expectedRemainingLiquidity := liquidityCreated.Sub(config.liquidityAmount)
+
+			// System under test.
+			amtDenom0, amtDenom1, err := concentratedLiquidityKeeper.WithdrawPosition(ctx, config.poolId, owner, config.lowerTick, config.upperTick, config.liquidityAmount)
+
+			if config.expectedError != nil {
+				s.Require().Error(err)
+				s.Require().Equal(amtDenom0, sdk.Int{})
+				s.Require().Equal(amtDenom1, sdk.Int{})
+				s.Require().ErrorAs(err, &config.expectedError)
```

Comment:
nit: don't need both error checks, last one is best

Reply:
The first one checks that the error is present and the last one checks the specific error returned. It's good practice to have two checks for ease of debugging.

e.g If nil is returned in place of error, I would now right away that the issue is beyond just the error type

Reply:
SGTM!